src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java

```diff
@@ -33,63 +33,79 @@ public ReadApartmentsXMLFormat() {
-                    : 0)
-            .setTerrace(prop.containsKey("terrace") && Boolean.valueOf(prop.getProperty("terrace")))
-            .setFloorAreaTerrace(
-                prop.containsKey("floorAreaTerrace")
-                    ? Double.parseDouble(prop.getProperty("floorAreaTerrace"))
-                    : 0)
-            .setDescription(prop.containsKey("description") ? prop.getProperty("description") : "")
-            .setTitle(prop.getProperty("title"))
-            .setWifi(prop.containsKey("wifi") ? Boolean.valueOf(prop.getProperty("wifi")) : false)
-            .setPricePerNight(
-                prop.containsKey("pricePerNight")
-                    ? Double.parseDouble(prop.getProperty("pricePerNight"))
-                    : 0)
-            .setNbMinNight(
-                prop.containsKey("nbMinNight")
-                    ? Integer.parseInt(prop.getProperty("nbMinNight"))
-                    : 0)
-            .setTele(prop.containsKey("tele") && Boolean.valueOf(prop.getProperty("tele")))
-            .build();
+    Builder apartBuilder = setObligatoryValues(new Builder());
+
+    if (prop.containsKey("floorAreaTerrace")) {
+      apartBuilder.setFloorAreaTerrace(Double.parseDouble(prop.getProperty("floorAreaTerrace")));
+    }
+    if (prop.containsKey("pricePerNight")) {
+      apartBuilder.setPricePerNight(Double.parseDouble(prop.getProperty("pricePerNight")));
+    }
+    if (prop.containsKey("description")) {
+      apartBuilder.setDescription(prop.getProperty("description"));
+    }
+    if (prop.containsKey("nbBedrooms")) {
+      apartBuilder.setNbBedrooms(Integer.parseInt(prop.getProperty("nbBedrooms")));
+    }
+    if (prop.containsKey("nbSleeping")) {
+      apartBuilder.setNbSleeping(Integer.parseInt(prop.getProperty("nbSleeping")));
+    }
+    if (prop.containsKey("nbBathrooms")) {
+      apartBuilder.setNbBathrooms(Integer.parseInt(prop.getProperty("nbBathrooms")));
+    }
+    if (prop.containsKey("nbMinNight")) {
+      apartBuilder.setNbMinNight(Integer.parseInt(prop.getProperty("nbMinNight")));
+    }
+
+    apartBuilder = setBooleanValues(apartBuilder);
+    Apartment apartment = apartBuilder.build();
 
     LOGGER.info("Parameters inserted with success in the Apartment Object");
     LOGGER.info("Leave readApartment method");
 
     return apartment;
   }
+
+  public Builder setObligatoryValues(Builder apartBuilder) {
+    if (prop.containsKey("address")) {
+      apartBuilder.setAddress(prop.getProperty("address"));
+    }
+    if (prop.containsKey("title")) {
+      apartBuilder.setTitle(prop.getProperty("title"));
+    }
+    if (prop.containsKey("floorArea")) {
+      apartBuilder.setFloorArea(Double.parseDouble(prop.getProperty("floorArea")));
+    }
+    return apartBuilder;
+  }
+
+  public Builder setBooleanValues(Builder apartBuilder) {
+    if (prop.containsKey("terrace")) {
+      apartBuilder.setTerrace(Boolean.valueOf(prop.getProperty("terrace")));
```

Comment:
Il ne serait pas mieux d'utiliser `.parseBoolean()` à la place de `.valueOf()` ?

Reply:
fix in c6be9ff98c1c38f413b86b273dab10185b34439c